shortestBridge.js

```diff
@@ -0,0 +1,187 @@
+/*
+We're going to approach this problem with a multi-source BFS. The grid will contain
+two islands, so we can first gather all the locations of the first island only.
+
+Once we have the locations of the first island, we can perform our BFS by adding 
+them to the queue with a starting path of 0. Since we're trying to find the shortest path 
+that connects with second island, starting with a path of 0 for all locations will be ideal 
+because with a BFS, you will always have the shorest path if you're visiting a 
+location for the first time.
+
+Once we reach a part of the second island, we will return the current path.
```

Comment:
This doesn't answer what I asked. Focus on this part: What if there was a shorter path, but it started from a different starting location?

Hint: see my answer to the previous comment.

Reply:
Let me know what you think:

It's possible to have an even shorter path from a different location but since we're starting from every location of the first island, we should the latest shortest path updated.

Reply:
Exactly. The key is that we are starting from every single location on the first island. Thus, it is impossible that there is some shorter path that we missed.